app/src/main/java/com/andrew/tasky/agenda/data/event/EventRepositoryImpl.kt

```diff
@@ -36,7 +37,27 @@ class EventRepositoryImpl @Inject constructor(
     private val eventPhotoCompressor = EventPhotoCompressor()
 
     override suspend fun upsertEvent(event: Event): UpsertEventResult {
-        val (newEvent, deletedPhotoCount) = compressAndDeletePhotosFromEvent(event)
+        val (compressedPhotos, deletedPhotoCount) = compressAndDeletePhotos(
+            event.photos.filterIsInstance<LocalPhoto>().filter {
+                !it.savedInternally
+            }
```

Comment:
Here you could use a named parameter to make it clear what this lambda evaluates to. That just reduces the time needed to think what this is for :)